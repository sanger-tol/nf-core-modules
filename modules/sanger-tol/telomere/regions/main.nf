process TELOMERE_REGIONS {
    tag "${meta.id}"
    label 'process_low'

    container 'quay.io/sanger-tol/telomere:0.0.1-c1'

    input:
    tuple val(meta), path(reference)
    val (telomereseq)

    output:
    tuple val( meta ), file( "*.telomere" ) , emit: telomere
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:

    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "TELOMERE_REGIONS module does not support Conda. Please use Docker / Singularity instead."
    }

    def prefix          = task.ext.prefix ?: "${meta.id}"
    def VERSION         = "1.0" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    find_telomere $reference $telomereseq > ${prefix}.telomere

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        find_telomere: ${VERSION}
    END_VERSIONS
    """

    stub:

    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "TELOMERE_REGIONS module does not support Conda. Please use Docker / Singularity instead."
    }

    def prefix          = task.ext.prefix ?: "${meta.id}"
    def VERSION         = "1.0" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    touch ${prefix}.telomere

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        find_telomere: ${VERSION}
    END_VERSIONS
    """

}
