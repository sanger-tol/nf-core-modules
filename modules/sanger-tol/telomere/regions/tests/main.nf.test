nextflow_process {

    name "Test Process TELOMERE_REGIONS"
    script "../main.nf"
    process "TELOMERE_REGIONS"

    tag "modules"
    tag "modules_sangertol"
    tag "telomere"
    tag "telomere/regions"

    test("GCA_963859965.1 - fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id: "Cloeon_Dipeterum" ],
                    file(params.modules_testdata_base_path + 'Cloeon_dipterum/assembly/release/ieCloDipt1.1/insdc/GCA_949628265.1.fasta.gz', checkIfExists: true)
                ]

                input[1] = Channel.value(
                    "TTAGGG"
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success               },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("GCA_963859965.1 - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: "Cloeon_Dipeterum" ],
                    file(params.modules_testdata_base_path + 'Cloeon_dipterum/assembly/release/ieCloDipt1.1/insdc/GCA_949628265.1.fasta.gz', checkIfExists: true)
                ]

                input[1] = Channel.value(
                    "TTAGGG"
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success               },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
