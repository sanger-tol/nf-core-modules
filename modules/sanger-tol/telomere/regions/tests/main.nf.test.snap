{
    "GCA_963859965.1 - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Cloeon_Dipeterum"
                        },
                        "Cloeon_Dipeterum.telomere:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd9100c4682ffb40fd48c62a94a9776a"
                ],
                "telomere": [
                    [
                        {
                            "id": "Cloeon_Dipeterum"
                        },
                        "Cloeon_Dipeterum.telomere:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd9100c4682ffb40fd48c62a94a9776a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-09T10:42:07.245827676"
    },
    "GCA_963859965.1 - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Cloeon_Dipeterum"
                        },
                        "Cloeon_Dipeterum.telomere:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd9100c4682ffb40fd48c62a94a9776a"
                ],
                "telomere": [
                    [
                        {
                            "id": "Cloeon_Dipeterum"
                        },
                        "Cloeon_Dipeterum.telomere:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd9100c4682ffb40fd48c62a94a9776a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-09T10:41:57.949126479"
    }
}